web-console/src/views/segments-view.tsx

```diff
@@ -133,8 +138,26 @@ export class SegmentsView extends React.Component<SegmentsViewProps, SegmentsVie
     });
   }
 
+  private initTableColumnSelection() {
+    if (localStorage.getItem(segmentTableColumnSelection) == null) {
+      const columns: string[] = ["Segment ID", "Datasource", "Start", "End", "Version", "Partition",
```

Comment:
It is really nasty that you repeat the columns here, columns should only be defined in one place the table column definitions. You should extract the table columns into their own `const` and then map out the values via some util function.